### ISSUE TITLE
Radiomaster Zorro

### ISSUE THREAD

**Issue by Spring7186** [bd4/edgetx-models#1]
Hello everybody, could these models file works on a Radiomaster Zorro? Thank you for answers.

**Comment by bd4** [comment]
It requires some work to migrate a model to a different model radio, because the switches are different. You can use EdgeTX companion to open the mode file with a Zorro radio configuration loaded, and you will get warnings and it may mess things up, but it still is arguably a lot easier than starting from scratch. You can also hand edit the .yml file to rename the switch names, but that can be pretty tedious. I wish there was a better solution. I wrote a Python script to help migrate some models but it's a bit of a mess, happy to share it if you are comfortable programming in Python but is not ready for general use.

**Comment by Spring7186** [comment]
Thank you for your very kind answer, unfortunately I'm not too much accustomed to Python, so I'll try with Companion hoping it will be not very difficult on switches reassigning operation.

**Comment by bd4** [comment]
I just did this converting some of my Boxer models to TX16S, and it was not bad. I did it like this:

1. Open companion and go to Settings->Radio Profiles, select target radio (in your case Zorro)
2. File->Open to open the .yml from a non-Boxer radio, will give a warning
3. Hand edit the model, check every switch usage (e.g. Input, Mixes, Flight modes, Logical Switches, Special Funcitons). Some of them are blank and obvious they need to be updated, but more tricky are ones that might be in a different location or behave differently (momentary vs 2pos vs 3pos) for example, so it's best to check every switch. You will need to change in multiple places - throttle cut switch for example (I used SD mostly) is in Logical Switches and Special Functions, and in some older models I made, might also be in Mixes. 
4. Plugin in target radio with usb storage mode, use "Read/Write->Read models and settings from radio" to load existing models, save a backup copy somewhere. Copy/paste the model you edited in (3) into the list you loaded from radio. Save a new copy, then do "Read/Write->Write models and setting to radio". Then eject USB drive on computer and unplug.

The main thing is to test thoroughly that is everything working the way you want on the ground, before putting anything in the air....

**Comment by Spring7186** [comment]
I will try what you suggested, for an Eachine E130 (should be similar to E129) should I modify your model no. 1 (Blade Nano S3)? Many, many thanks

**Comment by bd4** [comment]
I believe it uses the E129 protocol:
https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/Protocols_Details.md#E129---83

That is fairly different from the Nano S3 (the S3 has a separate pitch channel, E129/130 do not). It should be easier to setup than the S3 actually - you probably don't need any of the curves and don't need pitch on channel 5.

rcgroups.com and youtube may have some discussion threads / videos of people setting it up on one of the opentx/edgetx radios. Useful search terms along with "Eachine 130" are "opentx", "edgetx", "radiomaster", "tx16s". There is more content for tx16s than boxer or zorro specifically, and while the interface is slightly different on black and white vs color radios, all opentx and edgetx setups are pretty similar.

**Comment by Spring7186** [comment]
You are right, in fact I already downloaded the specific profile for my E130 from rcgroups, even if unfortunately it works for basics flight functionalities only. What I'm looking for, or at least what should I do to modify it to add others options of the original Eachine radio, such automatic take off/landing, emergency stop, trimmings, etc

**Comment by bd4** [comment]
The multiprotocol module details linked above should help with that, it lists Ch5 as takeoff/land, 6 as emergency, 7,8, 9 as trims. Doesn't say what hi/low do some still needs experimentation unfortunately.

**Comment by Spring7186** [comment]
I saw them, my problem is that I have no experience in creating the model

**Comment by bd4** [comment]
It really is intimidating at first. I highly recommend RC Video Reviews and Painless 360 on youtube, they have great EdgeTX instruction videos (and most OpenTX stuff is still relevant). RC Video Reviews also has a very active discord full of helpful people (including some edgetx and ExpressLRS developers), I think you may need to be a patreon to join but it's a couple dollars a month.

I think C189 has same protocol and settings, this may be a good starting point:
https://www.youtube.com/watch?v=dCK_8e278ws
he did not set up the landing feature though.

**Comment by Spring7186** [comment]
Yes, it could be a good starting point. Thank you very much indeed for your kind availability as per your useful suggestions.
